bookkeeper-server/pom.xml

```diff
@@ -182,6 +182,11 @@
       <artifactId>http-server</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```

Comment:
nit: I think we probably want to move the whole `checksum` package from pulsar to bookkeeper. but it is okay for this pull request. I will discuss with @merlimat on how to handle that.

Reply:
@sijie I'll address the comments later/tomorrow-ish.

On the other note I also tried/compared adler32 (from jdk) and crc32c from jdk9 (ported it to run with jdk8), neither were better than regular crc32.
oh, and I tried DUMMY digest (aka no digest). This one crazy fast and garbage free. Unfortunately, it is useless.

Reply:
@dlg99 did you run with jdk9? I've seem very good number with jdk9. in java 9, the crc32c with sse42 is intrinsic on jvm. I will send my results as well.

Reply:
In the checksum module we should be able to check for the JVM runtime version and select the fastest available implementation.

Reply:
@sijie I ran with jdk8. I tweaked http://hg.openjdk.java.net/jdk9/dev/jdk/file/feef46bbf70b/src/java.base/share/classes/java/util/zip/CRC32C.java to work on jdk8 and, to my surprise, saw no performance difference compared with jdk8's crc32.

Reply:
@dlg99 the approach you took didn't work because the assemble code of using intel crc32c instruction is a hotspot code which only available in jdk9. you can port the code from java9 to java8 and it can compile and work, but jdk8 doesn't this assemble code, so why you won't see any performance difference.

You can checkout the hotspot code here http://hg.openjdk.java.net/jdk9/jdk9/hotspot/rev/a60e232aa8f2#l11.20

Reply:
Also check my benchmark at #858

Reply:
crc32c will have its fullest potential only if it is using hardware instruction set. (obviously) :)

Reply:
Created #915 for tracking importing the pulsar-checksum module to bookkeeper as discussed with @merlimat before. 

It *doesn't* block this PR. This PR can go in first if we agreed with 4 bytes code length with crc32c.